windows/setup.ps1

```diff
@@ -17,15 +17,40 @@
 # WANT_JSON
 # POWERSHELL_COMMON
 
-# $params is not currently used in this module
-# $params = Parse-Args $args;
+# enabled $params (David O'Brien, 06/08/2015)
+$params = Parse-Args $args;
+
+
+Function Get-CustomFacts {
+  [cmdletBinding()]
+  param (
+    [Parameter(mandatory=$false)]
+    $factspath = $null
+  )
+
+  if (-not (Test-Path -Path $factspath)) {
+    Fail-Json $result "The path $factspath does not exist. Typo?"
+  }
+
+  $FactsFiles = Get-ChildItem -Path $factspath | Where-Object -FilterScript {($PSItem.PSIsContainer -eq $false) -and ($PSItem.Extension -eq '.ps1')}
+
+  foreach ($FactsFile in $FactsFiles) {
+      $out = . $($FactsFile.FullName)
+      Set-Attr $result.ansible_facts "ansible_$(($FactsFile.Name).Split('.')[0])" $out
+  }
+}
 
 $result = New-Object psobject @{
     ansible_facts = New-Object psobject
     changed = $false
 };
 
+If ($params.facts_path) {
```

Comment:
I think this will fail on Ansible 2.0 since it forces Powershell strict. I've modify this in a few modules. Check out Get-AnsibleParam in the powershell.ps1 file in module_utils